src/components/DpInput/DpInput.vue

```diff
@@ -37,248 +37,252 @@
+  label: {
+    type: Object,
+    default: () => ({
+      bold: true,
+      hide: false,
+      hint: '',
+      text: '',
+      tooltip: ''
+    }),
+    validator: (prop) => {
+      return Object.keys(prop).every(key => ['bold', 'hide', 'hint', 'text', 'tooltip'].includes(key))
     }
   },
 
-  methods: {
-    handleEnter (event) {
-      if (this.preventDefaultOnEnter === true) {
-        event.preventDefault()
-      }
-      this.$emit('enter')
+  /**
+   * Limit the maximum allowed number of characters to the given amount.
+   */
+  maxlength: length,
+
+  /**
+   * Define the minimum number of characters that need to be given.
+   */
+  minlength: length,
+
+  name: {
+    type: String,
+    required: false,
+    default: ''
+  },
+
+  pattern: {
+    type: String,
+    required: false,
+    default: ''
+  },
+
+  placeholder: {
+    type: String,
+    required: false,
+    default: ''
+  },
+
+  /**
+   * Set to false to prevent default behavior onEnter.
+   */
+  preventDefaultOnEnter: {
+    type: Boolean,
+    required: false,
+    default: true
+  },
+
+  readonly: {
+    type: Boolean,
+    required: false,
+    default: null
+  },
+
+  required: {
+    type: Boolean,
+    required: false,
+    default: null
+  },
+
+  /**
+   * When setting a number for the `size` prop, this is directly rendered
+   * as html attribute on the input element. Also, it is assumed that visual sizing
+   * based on that value shall be applied, that is why both container classes
+   * and element classes do not define any width styles when a size is set here.
+   */
+  size: {
+    type: [Number, null],
+    required: false,
+    default: null
+  },
+
+  type: {
+    type: String,
+    required: false,
+    default: 'text'
+  },
+
+  value: {
+    type: String,
+    required: false,
+    default: ''
+  },
+
+  /**
+   * Full width by default; set to 'auto' to have no width defined.
+   * @deprecated Apply width to the parent element of DpInput.
+   */
+  width: {
+    type: String,
+    default: 'w-full'
+  }
+})
+
+const emit = defineEmits(['blur', 'enter', 'focus', 'input'])
+
+const { value, maxlength, minlength, width, size, label } = toRefs(props)
+const currentValue = ref(value.value)
+
+const classes = computed(() => {
+  let cssClasses: string[] = [
+    'px-1 py-0.5 max-w-full rounded-input',
+    'text-base leading-4 bg-surface',
+    'outline outline-1 outline-offset-0 outline-transparent',
+    'focus-visible:outline-interactive focus-visible:border-interactive focus-visible:z-above-zero',
+    'required:shadow-none'
+  ]
+
+  if (!(size.value && size.value > 0)) {
+    cssClasses.push('w-full')
+  }
+
+  if (props.readonly || props.disabled) {
+    cssClasses.push('bg-surface-light border-none cursor-default')
+  } else {
+    cssClasses.push('text-input bg-surface border border-input cursor-text')
+  }
+
+  if (props.hasIcon) {
+    cssClasses.push('pr-4')
+  }
+
+  // o-form__control-search is nowhere styled directly
```

Comment:
Can this be removed?

Reply:
absolutely. 6bb17e6f31e49af5e6365b9e5a7cc5dfd89fcebb